### PR TITLE
Add CDK automation to run Credentials-fetcher in ECS

### PR DESCRIPTION
**Pull request by smhmhmd**
*Issue #, if available:*

*Description of changes:*
CDK automation to run gMSA in ECS with EC2 instance. This CDK can be used to test RPMs for AL2023.
This CDK does the following:
* Creates directory in Directory Service (Active Directory)
* Launch Windows instance, domain-join with Active Directory and create gMSA accounts
* Create ECS cluster
* Launch ECS-optimized Linux instance and attaches to ECS cluster
* Runs a couple of tasks in the ECS-optimized Linux instance

*Testing done:*
Yes, details of how to run test are in cdk/README.md

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [X] I have read the [CONTRIBUTING](https://github.com/aws/credentials-fetcher/blob/mainline/CONTRIBUTING.md) doc
- [X] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/credentials-fetcher/blob/mainline/CONTRIBUTING.md#commit-your-change)
- [X] I have updated any necessary documentation, including READMEs and comments (where appropriate)

#### Tests

- [X] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [X] I have checked that my tests are not configured for a specific environment

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by bhallasaksham — APPROVED** [review]

File: cdk/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+Pre-requisites
+1) Create secret in Secrets Manager as per https://docs.aws.amazon.com/AmazonECS/latest/developerguide/linux-gmsa.html#linux-gmsa-setup
+2) 'default' AWS profile
+
+Steps to run tasks in ECS with Credentials-fetcher.
```
Comment (as14692) [inline cdk/README.md:26]: Please mention whether this is domainless gMSA or domain-joined gMSA

File: cdk/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+Pre-requisites
+1) Create secret in Secrets Manager as per https://docs.aws.amazon.com/AmazonECS/latest/developerguide/linux-gmsa.html#linux-gmsa-setup
```
Comment (as14692) [inline cdk/README.md]: Please specify the corresponding secret to create used in the Test environment in `cdk/data.json`

File: cdk/data.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (as14692) [inline cdk/data.json:1]: Add note that these configurations are for test environment and not for production use

File: cdk/README.md (hunk 1 above)
Comment (smhmhmd) [reply cdk/README.md:26]: Will do

Comment (smhmhmd) [reply cdk/README.md:26]: Added in latest rev

File: cdk/README.md (hunk 2 above)
Comment (smhmhmd) [reply cdk/README.md]: Added in latest rev

File: cdk/data.json (hunk 3 above)
Comment (smhmhmd) [reply cdk/data.json:1]: Added in latest rev

File: cdk/README.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+Please take a look at data.json for default values.
+1) Create secret in Secrets Manager as per https://docs.aws.amazon.com/AmazonECS/latest/developerguide/linux-gmsa.html#linux-gmsa-setup with the following values:
+   This is the same secret in data.json.
+   ```
+    Secret key  Secret value
+    username    standarduser01
+    password    p@ssw0rd
+    domainName  activedirectory1.com
+    ```
+2) 'default' AWS profile with administrator access is needed, a separate/burner AWS account would suffice.
```
Comment (as14692) [inline cdk/README.md:24]: nit: Provide instructions to set up AWS CLI to run the `cdk` commands - https://docs.aws.amazon.com/cli/v1/userguide/cli-chap-configure.html

**Review by as14692 — APPROVED** [review]

**Comment by as14692** [comment]
Thank you so much for contributing the test infrastructure!

**Pull request merged**